### PR TITLE
Initial OpenDDS v3.14 changes

### PR DESCRIPTION
**Pull request by tpollock**
Modified OpenDDSharp to use OpenDDS v3.14.  There are still some issues with generating the type support files from Test.idl, which is causing build failures.

### CODE REVIEW

File: OpenDDSharp/OpenDDSharp.IdlGenerator/NativeIdlProject.vcxproj [hunk 1]
```diff
@@ -204,10 +204,10 @@
...
-    <Exec Command="cmd /c &quot;$(ACE_ROOT)\bin\tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I$(TAO_ROOT) -Sa -St -I$(DDS_ROOT) -Wb,export_macro=%(IdlFiles.Filename)IDL_Export -Wb,export_include=%(IdlFiles.Filename)IDL_Export.h &quot;%(IdlFiles.Identity)&quot;&quot;" IgnoreExitCode="false" ContinueOnError="false" />
-    <Exec Command="cmd /c &quot;$(DDS_ROOT)\bin\opendds_idl -Sa -St -Wb,export_macro=%(IdlFiles.Filename)IDL_Export &quot;%(IdlFiles.Identity)&quot;&quot;" IgnoreExitCode="false" ContinueOnError="false" />    
+    <Exec Command="cmd /c &quot;$(ACE_ROOT)\bin\tao_idl --idl-version 4 -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I$(TAO_ROOT) -Sa -St -I$(DDS_ROOT) -Wb,export_macro=%(IdlFiles.Filename)IDL_Export -Wb,export_include=%(IdlFiles.Filename)IDL_Export.h &quot;%(IdlFiles.Identity)&quot;&quot;" IgnoreExitCode="false" ContinueOnError="false" />
+    <Exec Command="cmd /c &quot;$(DDS_ROOT)\bin\opendds_idl --idl-version 4 -Sa -St -Wb,export_macro=%(IdlFiles.Filename)IDL_Export &quot;%(IdlFiles.Identity)&quot;&quot;" IgnoreExitCode="false" ContinueOnError="false" />
```
Comment (iguessthislldo) [inline OpenDDSharp/OpenDDSharp.IdlGenerator/NativeIdlProject.vcxproj:208]: `tao_idl` requires this flag, but `opendds_idl` defaults to idl 4, so it's not needed for it.

File: OpenDDSharp/OpenDDSharp.IdlGenerator/OpenDDSharp.IdlGenerator.targets [hunk 2]
```diff
@@ -24,7 +24,7 @@
   </Target>
   <Target Name="OpenDDSharpIdl" BeforeTargets="PreBuildEvent">
     <Message Text="Generating %(IdlFiles.Identity) code..." Importance="High" />
-    <Exec Command="$(MSBuildThisFileDirectory)..\tools\openddsharp_idl.exe &quot;%(IdlFiles.Identity)&quot;" IgnoreExitCode="false" ContinueOnError="false" />
+    <Exec Command="&quot;$(MSBuildThisFileDirectory)..\tools\openddsharp_idl.exe --idl-version 4 &quot;%(IdlFiles.Identity)&quot;&quot;" IgnoreExitCode="false" ContinueOnError="false" />
```
Comment (iguessthislldo) [inline OpenDDSharp/OpenDDSharp.IdlGenerator/OpenDDSharp.IdlGenerator.targets:27]: You're setting the idl version to 4 in OpenDDSharp/OpenDDSharp.IdlGenerator/be_init.cpp, so adding the flag here is redundant.

**Comment by jmmorato** [comment]
`IdlGenerator` must to be adapted to support the new IDLv4 annotations.

I have created a quick adaptation based on the OpenDDS changes [OpenDDSharp.IdlGenerator.zip](https://github.com/jmmorato/openddsharp/files/5142581/OpenDDSharp.IdlGenerator.zip).

Check it and let me know if it works for you.

File: Build/build.cmd [hunk 3]
```diff
@@ -7,20 +7,20 @@ IF EXIST "..\ext" (
...
-git checkout tags/DDS-3.13.2
+git checkout tags/DDS-3.14
 
 REM Apply the needed OpenDDS patches
-TITLE Apply the needed OpenDDS patches
-git apply %CD%\..\..\Patches\ConditionImpl.h.patch
-git apply %CD%\..\..\Patches\DataReaderImpl.cpp.patch
-git apply %CD%\..\..\Patches\DataReaderImpl.h.patch
+REM TITLE Apply the needed OpenDDS patches
+REM git apply %CD%\..\..\Patches\ConditionImpl.h.patch
```
Comment (jmmorato) [inline Build/build.cmd:21]: `ConditionImpl.h` should be still needed

File: Build/build.cmd [hunk 4]
```diff
@@ -7,20 +7,20 @@ IF EXIST "..\ext" (
...
+REM git apply %CD%\..\..\Patches\DataReaderImpl.cpp.patch
+REM git apply %CD%\..\..\Patches\DataReaderImpl.h.patch
     
 REM Download ACE/TAO and unzip
```
Comment (jmmorato) [inline Build/build.cmd:25]: We could use `--ace-github-latest` instead in the `configure` call

Comment (jmmorato) [inline Build/build.cmd:25]: We could use `--ace-github-latest` instead of the `Download ACE/TAO and unzip`

**Pull request closed without merging**